virtual_rainforest/models/soil/env_factors.py

```diff
@@ -0,0 +1,306 @@
+        clay_fraction: The clay fraction for each soil grid cell [unitless]
+        constants: Set of constants for the soil model
+
+    Returns:
+        An object containing four environmental factors, one for the effect of water
+        potential on enzyme rates, one for the effect of pH on enzyme rates, one for the
+        effect of clay fraction on enzyme saturation, and one for the effect of clay on
+        necromass decay destination.
+    """
+
+    # Calculate the impact that each environment variable has on the relevant
+    # biogeochemical soil processes
+    water_factor = calculate_water_potential_impact_on_microbes(
+        water_potential=soil_water_potential,
+        water_potential_halt=constants.soil_microbe_water_potential_halt,
+        water_potential_opt=constants.soil_microbe_water_potential_optimum,
+        moisture_response_curvature=constants.moisture_response_curvature,
+    )
+    pH_factor = calculate_pH_suitability(
+        soil_pH=pH,
+        maximum_pH=constants.max_pH_microbes,
+        minimum_pH=constants.min_pH_microbes,
+        lower_optimum_pH=constants.lowest_optimal_pH_microbes,
+        upper_optimum_pH=constants.highest_optimal_pH_microbes,
+    )
+    clay_factor_saturation = calculate_clay_impact_on_enzyme_saturation(
+        clay_fraction=clay_fraction,
+        base_protection=constants.base_soil_protection,
+        protection_with_clay=constants.soil_protection_with_clay,
+    )
+    clay_factor_decay = calculate_clay_impact_on_necromass_decay(
+        clay_fraction=clay_fraction,
+        decay_exponent=constants.clay_necromass_decay_exponent,
+    )
+
+    # Combine all factors into a single EnvironmentalFactors object
+    return EnvironmentalFactors(
+        water=water_factor,
+        pH=pH_factor,
+        clay_saturation=clay_factor_saturation,
+        clay_decay=clay_factor_decay,
+    )
+
+
+def calculate_temperature_effect_on_microbes(
+    soil_temperature: NDArray[np.float32],
+    activation_energy: float,
+    reference_temperature: float,
+) -> NDArray[np.float32]:
+    """Calculate the effect that temperature has on microbial metabolic rates.
+
+    This uses a standard Arrhenius equation to calculate the impact of temperature.
+
+    This function takes temperatures in Celsius as inputs and converts them into Kelvin
+    for use in the Arrhenius equation. TODO - review this after we have decided how to
+    handle these conversions in general.
+
+    Args:
+        soil_temperature: The temperature of the soil [C]
+        activation_energy: Energy of activation [J mol^-1]
+        soil_temperature: The reference temperature of the Arrhenius equation [C]
+
+    Returns:
+        A multiplicative factor capturing the effect of temperature on microbial rates
+    """
+
+    # Convert the temperatures to Kelvin
+    soil_temp_in_kelvin = convert_temperature(
+        soil_temperature, old_scale="Celsius", new_scale="Kelvin"
```

Comment:
This is something we need to agree on in the context of constants. I currently just add the conversion factor to the temperature in Celsius, I am not sure if it needs a function for that simple operation.

Reply:
ok, I see that the C2K is deprecated, we can chat about this tomorrow

Reply:
It's a pity it was deprecated, because it's a lot terser! I suppose you could write your own helper function though?

Reply:
@vgro per our discussion the other day I've made an issue about switching to using Kelvin as the default unit #346. That should minimise the number of conversions that have to be made throughout the model